### PR TITLE
Changes to handle device deregsitration via provisioning an empty "cis"

### PR DESCRIPTION
**Pull request by marktrayer**

### CODE REVIEW

**Review by ondrejtomcik — CHANGES_REQUESTED** [review]
@marktrayer why did you drop our ttl related change?

**Comment by marktrayer** [comment]
> 
> 
> @marktrayer why did you drop our ttl related change?

That's an accident; I'll fix that.

File: api/cloud/oc_cloud_resource.c
Comment (jkralik) [inline api/cloud/oc_cloud_resource.c:167]: this line creates a body for response. why did you commented it?

File: include/oc_cloud.h
Comment (jkralik) [inline include/oc_cloud.h:135]: pls don't drop TTL

**Review by jkralik — CHANGES_REQUESTED** [review]

File: api/cloud/oc_cloud_apis.c
Comment (jkralik) [inline api/cloud/oc_cloud_apis.c:279]: pls use `cloud_set_cps()` . Otherwise you will not notify observers.

File: api/cloud/oc_cloud.c
Comment (jkralik) [inline api/cloud/oc_cloud.c:144]: pls leave this line - it will save cloud store to storage(file).

**Review by jkralik — CHANGES_REQUESTED** [review]
Pls fix opened comments. Could you format changes with clang-format (https://github.com/iotivity/iotivity-lite/blob/master/tools/README) ?

**Review by jkralik — APPROVED** [review]

**Review by ondrejtomcik — APPROVED** [review]

File: api/cloud/oc_cloud.c
Comment (jkralik) [inline api/cloud/oc_cloud.c:196]: I think we need to call `oc_cloud_deregister` otherwise the cloud is not informed about deregistration.

**Review by jkralik — CHANGES_REQUESTED** [review]
cloud needs to be informed about deregistration.

File: api/cloud/oc_cloud.c
Comment (marktrayer) [reply api/cloud/oc_cloud.c:196]: oc_cloud_reset_context does a deregistration within the method call.

Comment (jkralik) [reply api/cloud/oc_cloud.c:196]: Oh. I didn't expect that.

**Review by jkralik — APPROVED** [review]

**Review by ondrejtomcik — APPROVED** [review]

File: api/cloud/oc_cloud_apis.c
Comment (Danielius1922) [inline api/cloud/oc_cloud_apis.c:279]: [FYI] there seems to be a bit of inconsistent formatting in these changes, no reason to block merge because of that, but its something I've noticed. 

There is a clang-format files in the [tools folder](https://github.com/iotivity/iotivity-lite/tree/master/tools), so formatting can be automated and enforced. @WAvdBeek whatabout running clang-format over the whole repo and adding a workflow to check formatting in GitHub?

Comment (WAvdBeek) [reply api/cloud/oc_cloud_apis.c:279]: it was disabled since it was taking more time to fix formatting than the actual code.

Comment (Danielius1922) [reply api/cloud/oc_cloud_apis.c:279]: I am confused, does the formatting need corrections or something? Because it can be fully automated by installing the pre-commit hook locally, clang will handle the formatting automatically then and nothing needs to be done by the programmer.

Comment (WAvdBeek) [reply api/cloud/oc_cloud_apis.c:279]: no worries about the formatting, currently not checked.

next week I will try to get (someone) the automatic formatter to work on the code base.

**Comment by WAvdBeek** [comment]
Mark, this should go into the Jakarta branch, not Master..

or we wait until this work is finished (note i like to make a 2.2.3.1 for we start to merge jakarta work into main)

**Review by Danielius1922 — APPROVED** [review]

**Comment by Danielius1922** [comment]
I've rebased this PR it get conflicts in files that changed after my PR got merged, hopefully I've resolved the conflicts correctly, but I think we should verify if it didn't break the bugfix.

**Review by ondrejtomcik — APPROVED** [review]

**Pull request merged**
